README.md

```diff
@@ -109,13 +109,17 @@ This allows users to do advanced things such as wrapping up objectives in
 a more complex way, using callbacks, controlling termination, optimising after N suggestions,
 continuing iterating if solution is not satisfactory, and so on.
 
+One should call `Graph.checkspace(space)` prior to using `ask` -- to avoid inefficiency of repeatedly
+checking the space is valid for each ask the user is required to do this themselves.
```

Comment:
I am don't think that is this change is related to the main purposes of this PR.

Reply:
No its just that it caused me a problem while developing the tests, so I figured its probably a good guidance to provide directly to users.

Reply:
Fair enough but might want to put this in separate commit.